Web3Swift/Network/SimpleNetwork.swift

```diff
@@ -1,20 +1,79 @@
 import Foundation
 import SwiftyJSON
 
+internal final class IncorrectUrlStringError: DescribedError {
+
+    private let url: String
+
+    init(url: String) {
+        self.url = url
+    }
+
+    internal var description: String {
+        return "Incorrect URL string \"\(self.url)\""
+    }
+
+}
+
+internal final class InvalidIDResponseError: DescribedError {
+
+    private let response: String
+    init(response: String) {
+        self.response = response
+    }
+
+    var description: String {
+        return "net_version call was expected to return \"result\" as decimal in a string but it was \(response)"
+    }
+
+}
+
 public class SimpleNetwork: Network {
     
-    private var session: URLSession
-    private var url: URL
-    private var headers: Dictionary<String, String>
+    private let session: URLSession
+    private let url: String
+    private let headers: Dictionary<String, String>
     
-    init(session: URLSession, url: URL, headers: Dictionary<String, String>) {
+    init(session: URLSession, url: String, headers: Dictionary<String, String>) {
         self.session = session
         self.url = url
         self.headers = headers
     }
 
+    /**
+    - returns:
+    id of a network
+
+    - throws:
+    `DescribedError` if something went wrong.
+    */
+    public func id() throws -> NumberScalar {
```

Comment:
@biboran each `procedure` depends on `network` and we use `procedure` in one of the `network` implementations.

Reply:
@rockfridrich I believe there is nothing wrong with that.

Reply:
@biboran ok, but let's think of introducing some sort of `Chain` object for this purposes later.